product_code_builder_sequence/models/product_product.py

```diff
@@ -46,3 +46,18 @@ def copy(self, default=None):
                     'prefix_code': self.prefix_code + _('-copy'),
                 })
         return super(ProductTemplate, self).copy(default)
+
+
+class ProductProduct(models.Model):
+    _inherit = 'product.product'
+
+    @api.multi
+    def copy(self, default=None):
+        for product in self:
+            if default is None:
+                default = {}
+            if product.prefix_code:
```

Comment:
Note @StefanRijnhart this is a direct copy of the existing product.template code, so the existing module has the same error.  

@espo-tony In addition the existing module also has the error of product.prefix_code morphing into self.prefix_code on the following line.